src/compiler/checker.ts

```diff
@@ -4753,6 +4753,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         }
                     }
                 }
+
+                if (moduleNotFoundError) {
+                    if (tryGetExtensionFromPath(sourceFile.fileName) === tryGetExtensionFromPath(moduleReference)) {

```

Comment:
Doesn’t this falsely trigger on `import {} from "./foo.ts"` when the source file is `foo.ts.ts`? In my implementation, I couldn’t find a better way than to add a property onto `ResolvedModule` that indicated whether the `.ts` extension was actually used to match a `.ts` file extension.

Reply:
The error says that the "import path can't end in `.ts`" not that it can't _resolve_ to TS, so I'd argue it's not a false positive. Honestly, a `"./foo.ts"` that resolves to `"./foo.ts.js"` is expempted from the error right now, but I'm questioning more if _that's_ OK - blanket saying "relative import specifiers can't end in .ts unless you pass some flag" just seems easier to understand, and covers weird cjs corners like this, if overzealously.